### PR TITLE
Support abiv2

### PR DESCRIPTION
**Pull request by josh-richardson**
### What does this PR do?
*required*

### Where should the reviewer start?
*required*

### Why is it needed?
*required*

### CODE REVIEW

File: codegen/src/main/java/org/web3j/codegen/SolidityFunctionWrapper.java [hunk 1]
```diff
@@ -1385,18 +1470,40 @@ CodeBlock buildTypedResponse(
...
+            List<AbiDefinition.NamedTypeComponent> components,
+            boolean primitives)
             throws ClassNotFoundException {
+        if ("tuple".equals(typeDeclaration)) {
+            // There must be a components argument!
+            if (components == null || components.size() == 0) {
+                throw new IllegalArgumentException("Components must be provided for 'tuples'");
+            }
+
+            // TODO[Sam]: sort out package name as parameter.
```
Comment (cfelde) [inline codegen/src/main/java/org/web3j/codegen/SolidityFunctionWrapper.java]: TODOs

Comment (cfelde) [reply codegen/src/main/java/org/web3j/codegen/SolidityFunctionWrapper.java]: Never mind, I see it's a draft PR

**Comment by iikirilov** [comment]
@josh-richardson are you working on this?

**Comment by iikirilov** [comment]
replaced by #1206

**Pull request closed without merging**